flake.nix

```diff
@@ -0,0 +1,75 @@
+{
+  description = "Agoric SDK Development Environment";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+               # Custom Node.js 20.9.0 package
+        custom_node_20_9 = pkgs.stdenv.mkDerivation {
```

Comment:
why do we need a custom node package? I can see that we do -- I got `error: undefined variable 'nodejs-20_x'` without it. But... what's going on? Why didn't it Just Work?

When we move to node 22, how do we maintain this?

Reply:
@dckc so what happens is that I need to ensure we download the right version of node, So I make my own derivation if we change node version I need to update this, I can find ways to keep this slightly more automatic, Also maybe lets jump on a call to discuss this

Reply:
> I need to ensure we download the right version of node

I'm struggling to understand why. Don't the nixpkgs maintainers ensure that the right version of node is downloaded?

Reply:
ok... so this is "custom" from a nixpkgs perspective, but from the perspective of nodejs.org, this is the standard version.

Reply:
This is correct